src/plugins/irc/irc-command.c

```diff
@@ -237,6 +238,133 @@ irc_command_mode_nicks (struct t_irc_server *server,
 }
 
 /*
+ * Sends mode change for many masks on a channel.
+ *
+ * Argument "set" is "+" or "-", mode can be "b", "q", or any other mode
+ * supported by server.
+ *
+ * Many messages can be sent if the number of nicks is greater than the server
+ * limit (number of modes allowed in a single message). In this case, the first
+ * message is sent with high priority, and subsequent messages are sent with low
+ * priority.
+ */
+
+void
+irc_command_mode_masks (struct t_irc_server *server,
+                        const char *channel_name,
+                        const char *command,
+                        const char *set, const char *mode,
+                        char **argv, int pos_masks)
+{
+    int max_modes, modes_added, msg_priority;
+    long number;
+    char *error, modes[128+1], masks[1024], *mask;
+    const char *ptr_modes;
+    struct t_irc_channel *ptr_channel;
+    struct t_irc_nick *ptr_nick;
+
+    if (irc_mode_get_chanmode_type (server, mode[0]) != 'A')
+    {
+        weechat_printf (
+            NULL,
+            _("%s%s: cannot use /%s, type A channel mode \"%s\" not supported "
+              "by server"),
+            weechat_prefix ("error"), IRC_PLUGIN_NAME, command, mode);
+        return;
+    }
+
+    /* default is 4 modes max (if server did not send the info) */
+    max_modes = 4;
```

Comment:
Maybe 3 is a better default since that's in [RFC 2812](https://tools.ietf.org/html/rfc2812#section-3.2.3).